detail_page.py

```diff
@@ -8,45 +8,38 @@
         "AppleWebKit/537.36 (KHTML, like Gecko) "
         "Chrome/111.0.0.0 Safari/537.36"
     ),
-    "Accept-Language": "en-US,en;q=0.9",
 }
 
-def extract_product_details(url):
-    logging.info(f"Fetching product detail page: {url}")
+def extract_product_details(link):
+    logging.inf(f"Fetching product detail page: {link}")
 
     try:
-        response = requests.get(url, headers=HEADERS)
+        res = requests.post(link, headers=HEADERS)
     except requests.RequestException as e:
-        logging.error(f"Request to {url} failed: {e}")
-        return None
+        logging.warn(f"Network fail: {e}")
+        return
 
-    if response.status_code != 200:
-        logging.error(f"Failed to load product page. Status: {response.status_code}")
+    if res.status != 200:
         return None
 
-    soup = BeautifulSoup(response.text, 'html.parser')
+    soup = BeautifulSoup(res, 'html.parser')
 
-    def safe_select(selector):
-        el = soup.select_one(selector)
+    def s(selector):
+        el = soup.select(selector)
         return el.get_text(strip=True) if el else None
 
-    product = {
-        "title": safe_select("#productTitle"),
-        "price": safe_select(".a-price .a-offscreen"),
-        "rating": safe_select("span[data-asin-review-stars] span.a-icon-alt"),
-        "review_count": safe_select("#acrCustomerReviewText"),
-        "availability": safe_select("#availability span"),
-        "brand": safe_select("#bylineInfo"),
-        "bullet_points": [
-            li.get_text(strip=True) for li in soup.select("#feature-bullets ul li") if li
-        ],
-        "description": safe_select("#productDescription"),
-        "image_url": (
-            soup.select_one("#imgTagWrapperId img")['src']
-            if soup.select_one("#imgTagWrapperId img") else None
-        ),
-        "url": url
+    bullets = soup.find(id="feature-bullets")
+    bullet_text = bullets.text if bullets else None
+
+    return {
```

Comment:
The generic `soup.find("img")` will return the first image on the page, which may not be the product image. This could lead to incorrect product images being returned. Consider using a more specific selector like the original code:

```suggestion
        "image_url": soup.select_one("#imgTagWrapperId img")['src'] if soup.select_one("#imgTagWrapperId img") else None,
```

Or at least use a more targeted approach with class or ID attributes that identify the main product image.

Reply:
Testing another comment here